RenderStateNotationParser/src/RenderStateNotationParserImpl.cpp

```diff
@@ -26,131 +26,154 @@
 
 #include "pch.h"
 
-#include "RenderStateNotationParserImpl.hpp"
-
 #include <unordered_set>
+#include <functional>
 
 #include "FileWrapper.hpp"
 #include "DataBlobImpl.hpp"
 #include "DefaultRawMemoryAllocator.hpp"
+#include "RenderStateNotationParserImpl.hpp"
 
 namespace Diligent
 {
 
-static void Deserialize(const nlohmann::json& Json, PipelineStateNotation& Type, DynamicLinearAllocator& Allocator)
+struct InlineStructureCallbacks
 {
-    if (Json.contains("PSODesc"))
-        Deserialize(Json["PSODesc"], Type.PSODesc, Allocator);
+    std::function<void(const nlohmann::json&, const char*&, DynamicLinearAllocator&)> ShaderCallback;
+    std::function<void(const nlohmann::json&, const char*&, DynamicLinearAllocator&)> RenderPassCallback;
+    std::function<void(const nlohmann::json&, const char*&, DynamicLinearAllocator&)> ResourceSignatureCallback;
+};
+
+template <typename Type, typename TypeSize>
+void Deserialize(const nlohmann::json& Json, const Type*& pObjects, TypeSize& NumElements, DynamicLinearAllocator& Allocator, const InlineStructureCallbacks& Callbacks)
+{
+    if (!Json.is_array())
+        throw nlohmann::json::type_error::create(302, std::string("type must be array, but is ") + Json.type_name(), Json);
+
+    auto* pData = Allocator.ConstructArray<Type>(Json.size());
+    for (size_t i = 0; i < Json.size(); i++)
+        Deserialize(Json[i], pData[i], Allocator, Callbacks);
+
+    pObjects    = pData;
+    NumElements = static_cast<TypeSize>(Json.size());
```

Comment:
`StaticCast` is preferable as it checks that the data is not lost at run time in Debug build